src/OrchardCore.Modules/OrchardCore.Metadata/Drivers/MetadataPartDisplayDriver.cs

```diff
@@ -0,0 +1,244 @@
+                            Content = viewModel.OpenGraphDescription
+                        });
+                    }
+                    if (!String.IsNullOrWhiteSpace(viewModel.OpenGraphImage))
+                    {
+                        _resourceManager.RegisterMeta(new MetaEntry
+                        {
+                            Property = "og:image",
+                            Content = viewModel.OpenGraphImage
+                        });
+                    }
+                }
+                // Extended metatags for Facebook Open Graph
+                if (settings.SupportOpenGraph)
+                {
+                    if (!String.IsNullOrWhiteSpace(viewModel.OpenGraphType))
+                    {
+                        _resourceManager.RegisterMeta(new MetaEntry
+                        {
+                            Property = "og:type",
+                            Content = viewModel.OpenGraphType
+                        });
+                    }
+                    if (!String.IsNullOrWhiteSpace(viewModel.OpenGraphUrl))
+                    {
+                        _resourceManager.RegisterMeta(new MetaEntry
+                        {
+                            Property = "og:url",
+                            Content = viewModel.OpenGraphUrl
+                        });
+                    }
+                    if (!String.IsNullOrWhiteSpace(viewModel.OpenGraphImageAlt) && !String.IsNullOrWhiteSpace(viewModel.OpenGraphImage))
+                    {
+                        _resourceManager.RegisterMeta(new MetaEntry
+                        {
+                            Property = "og:image:alt",
+                            Content = viewModel.OpenGraphImageAlt
+                        });
+                    }
+                }
+                // Extended metatags for Twitter
+                if (settings.SupportTwitterCards)
+                {
+                    if (!String.IsNullOrWhiteSpace(viewModel.TwitterCard))
+                    {
+                        _resourceManager.RegisterMeta(new MetaEntry
+                        {
+                            Name = "twitter:card",
+                            Content = viewModel.TwitterCard
+                        });
+                    }
+                    // Twitter does not support the Open Graph Image Alt tag so we have to add Twitter's own tag
+                    if (!String.IsNullOrWhiteSpace(viewModel.OpenGraphImageAlt) && !String.IsNullOrWhiteSpace(viewModel.OpenGraphImage))
+                    {
+                        _resourceManager.RegisterMeta(new MetaEntry
+                        {
+                            Name = "twitter:image:alt",
+                            Content = viewModel.OpenGraphImageAlt
+                        });
+                    }
+                }
+            }
+
+            // TODO these should be added on every page regardless of whether the Metadata part is added to the content type
```

Comment:
This TODO can be done inside a filter. We have an example for notifications (NotifyFilter I think).